Mathlib/Geometry/Manifold/ContMDiff/Product.lean

```diff
@@ -347,3 +347,156 @@ theorem contMDiff_pi_space :
 @[deprecated (since := "2024-11-20")] alias smooth_pi_space := contMDiff_pi_space
 
 end PiSpace
+
+section disjointUnion
+
+variable {M' : Type*} [TopologicalSpace M'] [ChartedSpace H M'] {n : WithTop ℕ∞} [Nonempty H]
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace 𝕜 E'] {H' : Type*} [TopologicalSpace H']
+  {J : Type*} {J : ModelWithCorners 𝕜 E' H'}
+  {N N' : Type*} [TopologicalSpace N] [TopologicalSpace N'] [ChartedSpace H' N] [ChartedSpace H' N']
+
+open Topology
+
+lemma ContMDiff.inl : ContMDiff I I n (@Sum.inl M M') := by
+  intro x
+  rw [contMDiffAt_iff]
+  refine ⟨continuous_inl.continuousAt, ?_⟩
+  -- In extended charts, .inl equals the identity (on the chart sources).
+  apply contDiffWithinAt_id.congr_of_eventuallyEq; swap
+  · simp [ChartedSpace.sum_chartAt_inl]
+    congr
+    apply Sum.inl_injective.extend_apply (chartAt _ x)
+  set C := chartAt H x
```

Comment:
Inlining this `set` makes the following have leave unsolved goals. I'm not fully sure why.